### ISSUE TITLE
can't load react-globalize in vendor bundle

### ISSUE THREAD

**Issue by alunny** [rxaviers/globalize-webpack-plugin#8]
This is a bit involved, so please bear with me :)

In our app, we use default messages for calls to `messageFormatter`. These look like:

```js
const helloMsg = Globalize.messageFormatter('Hello {name}');
```

When react-globalize is imported in the module (or otherwise in the bundle), it patches `messageFormatter` to escape these strings, so the English message looks like:

```json
{
  "en": {
    "Hello (name)": "Hello {name}"
  }
}
```

When react-globalize is not loaded, the phrase is extracted as `"Hello {name}"`, which doesn't match the message and causes a runtime error.

To fix this, we tried putting `react-globalize` inside the vendor bundle, to ensure it is always loaded. This worked perfectly in development mode.

Unfortunately, in production, `globalize-webpack-plugin` rewrites all of the globalize module imports to refer to the i18n bundle, which is in a separate file. So with the following files:

```
vendor.js
i18n.en.js
main.js
```

react-globalize's imports are rewritten to refer to the patched globalize in `i18n.en.js`, not the original Globalize module in `vendor.js`, which fails at runtime.

I think the best solution would be for `globalize-webpack-plugin` to provide an exclusion mechanism for modules which should not have their module imports rewritten (similar to `util.isGlobalizeModule`). Then `react-globalize-webpack-plugin` could specify not to rewrite the react-globalize modules, and it could correctly live in vendor.js with the original Globalize import.

If this sounds good to you @rxaviers, I can get started on the patches (one here, one in react-globalize-webpack-plugin).

**Comment by rxaviers** [comment]
The first issue is that the default messages concept is implemented by `react-globalize`. Ideally, we should move it into `globalize` directly.

Your exclusion solution sounds good to me.

**Comment by rxaviers** [comment]
Released as 0.3.0.

PS: react-globalize-webpack-plugin has been updated as well.
